lib/msf/core/exploit.rb

```diff
@@ -360,6 +360,56 @@ def has_auto_target?(targets=[])
   #
   ##
 
+  #
+  # The check method performs a benign attack against the target.  Ideally,
+  # it will be overwritten by the exploit module with a more specific
+  # vulnerability-specific technique.  However, if the module is not over-
+  # written, let's use the exploit method, alongside a pingback payload,
+  # to determine if the target is vulnerable.
+  #
+
+  def check
+    first_target = find_target_supporting_pingback_payload
+    if first_target
+      datastore['TARGET'] = first_target
+      vprint_status "Found a target that supports pingback payloads: #{first_target} (#{targets[first_target].name})"
+
+      datastore['PAYLOAD'] = find_pingback_payload_in_target
+      vprint_status "Configuring a pingback payload: #{self.payload_instance}"
+      generate_payload
+
+      exploit
+    else
+      vprint_error "No targets within this module support pingback payload."
+      return false
+    end
+  end
+
+  def find_target_supporting_pingback_payload
+    k = 0
+    targets.each do |target|
+      datastore['TARGET'] = k
```

Comment:
This will overwrite any target set by the user even a suitable one to check with isn't found.